src/nets/e2e_asr_attctc_th.py

```diff
@@ -484,21 +583,15 @@ def forward(self, hpad, hlen, ys):
             z_all.append(z)
             att_weight_all.append(att_w.data)  # for debugging
 
-        z_all = torch.stack(z_all, dim=1).view(batch * olength, self.dunits)  # NOTE: maybe cat?
+        z_all = torch.stack(z_all, dim=1).view(batch * olength, self.dunits)
         # compute loss
         y_all = self.output(z_all)
-        # NOTE: use size_average=True?
         self.loss = torch.nn.functional.cross_entropy(y_all, pad_ys_out.view(-1),
-                                                      ignore_index=ignore_id, size_average=True)
-        # NOTE: is this length-scaling required?
+                                                      ignore_index=self.ignore_id, size_average=True)
         # -1: eos, which is removed in the loss computation
         self.loss *= (np.mean([len(x) for x in ys_in]) - 1)
         # acc = F.accuracy(y_all, F.concat(pad_ys_out, axis=0), ignore_label=-1)
-        acc = 0
-        pred_pad = y_all.data.view(len(ys), olength, y_all.size(1)).max(2)[1]
-        for i in range(len(ys)):
-            acc += torch.sum(pred_pad[i, :ys[i].size(0)] == ys[i].data)
-        acc /= sum(map(len, ys))
+        acc = th_accuracy(y_all, pad_ys_out, ignore_label=self.ignore_id)
         logging.info('att loss:' + str(self.loss.data))
 
         # show predicted character sequence for debug
```

Comment:
In the case of chainer, ignore label is `-1`.
Therefore, we should change as follows.
```
# now 
                 idx_hat = np.argmax(y_hat_[y_true_ != -1], axis=1)
                 idx_true = y_true_[y_true_ != -1]
# proposed
                 idx_hat = np.argmax(y_hat_[y_true_ != self.ignore_id], axis=1)
                 idx_true = y_true_[y_true_ != self.ignore_id]
```

Reply:
OK. I think you are talking about `show predicted character sequence for debug` at Decoder.forward.

Reply:
Yes, that's right.

Reply:
Sorry, but I noticed the current version self.ignore_id=0. Is 0 for CTC?

Reply:
Are you talking about this?
https://github.com/espnet/espnet/blob/c044f7a25dad63282071a38cef5b4678f9158fc3/src/nets/e2e_asr_attctc_th.py#L1633
This may have some problems.
@ShigekiKarita, can you tell me why you set 0 here?

Reply:
In current implementation, character index for decoder is 1 to odim - 1.
Therefore, 0 is not used in decoder and it might be no effect even if we use 0 as ignore id, I think.

Reply:
But to make more clear, I agree to change it to -1.

Reply:
@kan-bayashi Many thanks for your reply!